### ISSUE TITLE
[ECS] [request]: support HTTPS App Protocol

### ISSUE THREAD

**Issue by stdevi** [aws/containers-roadmap#2428]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
When setting up Service Connect you can optionally set a field for `appProtocol` in the task definition, https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html.

Valid protocol values are "HTTP", "HTTP2", and "GRPC". I would like to request the addition of support for "HTTPS" as an application protocol in AWS Service Connect.

**Which service(s) is this request for?**
ECS, Service Connect.

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Currently, when using Service Connect, the available application protocols are limited to HTTP, HTTP2, and gRPC, with no support for HTTPS. As a result, applications that require HTTPS must implement workarounds, such as using load balancers to handle HTTPS traffic.

**Are you currently working around this issue?**
I am using load balancers to handle HTTPS traffic, which adds complexity and overhead to the deployment process.

**Comment by coultn** [comment]
Service Connect supports TLS (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-connect-tls.html), does that not meet your needs?

**Comment by stdevi** [comment]
@coultn thank you for the suggestion! If we use Service Connect with TLS and set AppProtocol to `http`, does that require opening an unencrypted HTTP port? If so, how does this configuration align with TLS encryption and compliance with standards like SOC 2 and HIPAA?

**Comment by tamir-deep** [comment]
@coultn From what I tested with SC TLS, the services still must communicate between them using the HTTP protocol. Switching over to HTTPS throws an generic OpenSSL error.

**Comment by kshivaz** [comment]
Regarding TLS, inbound and outbound traffic passing through the ECS Service Connect Proxy is encrypted and uses HTTPS. However, communication between the ECS Service Connect Proxy and the container happens in plain text, with the server continuing to listen on a HTTP port. This is why you observe services communicating over HTTP locally. This model is designed to be highly secure, as all communication that traverses the network is encrypted.

You can verify TLS is enabled for your services by following the steps outlined in the AWS documentation - https://docs.aws.amazon.com/AmazonECS/latest/developerguide/verify-tls-enabled.html

**Comment by tamir-deep** [comment]
@kshivaz Yeah I indeed verified it was encrypted using this guide and it works.

The only concern for this use case is the fact the container must still listen on HTTP instead of HTTPS which sometimes is a must for regulatory reasons in some companies/sectors like banking.

Nevertheless, It's still a good use case for us.
